invisible_cities/cities/base_cities.py

```diff
@@ -259,16 +259,32 @@ def set_up_database(self):
         self.DataPMT  = DataPMT
         self.DataSiPM = DataSiPM
 
-        self.xs                 = DataSiPM.X.values
-        self.ys                 = DataSiPM.Y.values
-        self.pmt_active         = np.nonzero(pmt_active)[0].tolist()
-        self.active_pmt_ids     = DataPMT.SensorID[DataPMT.Active == 1].values
-        self.all_pmt_adc_to_pes = abs(DataPMT.adc_to_pes.values).astype(np.double)
-        self.    pmt_adc_to_pes = self.all_pmt_adc_to_pes[pmt_active]
-        self.   sipm_adc_to_pes = DataSiPM.adc_to_pes.values    .astype(np.double)
-        self.coeff_c            = DataPMT.coeff_c.values        .astype(np.double)
-        self.coeff_blr          = DataPMT.coeff_blr.values      .astype(np.double)
-        self.noise_rms          = DataPMT.noise_rms.values      .astype(np.double)
+        self.xs                  = DataSiPM.X.values
+        self.ys                  = DataSiPM.Y.values
+        self.pmt_active          = np.nonzero(pmt_active)[0].tolist()
+        self.active_pmt_ids      = DataPMT.SensorID[DataPMT.Active == 1].values
+        self.all_pmt_adc_to_pes  = abs(DataPMT.adc_to_pes.values).astype(np.double)
+        self.    pmt_adc_to_pes  = self.all_pmt_adc_to_pes[pmt_active]
+        self.   sipm_adc_to_pes  = DataSiPM.adc_to_pes.values    .astype(np.double)
+        self.coeff_c             = DataPMT.coeff_c.values        .astype(np.double)
+        self.coeff_blr           = DataPMT.coeff_blr.values      .astype(np.double)
+        self.noise_rms           = DataPMT.noise_rms.values      .astype(np.double)
+
+        ## Charge resolution for sensor simulation
+        pmt_single_pe_rms      = DataPMT.Sigma.values          .astype(np.double)
+        self.pmt_pe_resolution = np.divide(pmt_single_pe_rms                         ,
+                                           self.all_pmt_adc_to_pes                   ,
+                                           out=np.zeros(len(self.all_pmt_adc_to_pes)),
+                                           where=self.all_pmt_adc_to_pes!=0          )
+        sipm_single_pe_rms     = DataSiPM.Sigma.values         .astype(np.double)
+        if not sipm_single_pe_rms.any():
```

Comment:
Shouldn't this be handled by the database?

Reply:
In principle it could be, I'll look into it but it might imply more changes than this implementation